### PR TITLE
Avoid BC break on cache fix

### PR DESCRIPTION
**Pull request by erikn69**
Last try
Using laravel's `hydrate` for avoid changes on interface, 
i've never used it before but it seems to work
Better solution for #1799 problem

### CODE REVIEW

File: src/PermissionRegistrar.php [hunk 1]
```diff
@@ -123,18 +123,19 @@ private function loadPermissions()
...
             if (is_array($this->permissions)) {
-                $this->permissions = (new Collection($this->permissions))->map(function ($permission) {
-                    return $this->permissionClass::getModelFromArray($permission);
+                $this->permissions = $this->getPermissionClass()::hydrate(
+                    collect($this->permissions)->map(function ($item) {
+                        return collect($item)->only('id', 'name', 'guard_name')->all();
+                    })->all()
+                )
+                ->each(function ($permission, $i) {
+                    $permission->setRelation('roles', $this->getRoleClass()::hydrate($this->permissions[$i]['roles'] ?? []));
```
Comment (drbyte) [inline src/PermissionRegistrar.php:138]: Does this make it as "small" as before? I think it leaves the `['pivot']` sub-array, doesn't it?

Comment (erikn69) [reply src/PermissionRegistrar.php:138]: I don't think that, `hydrate` makes a Collection from array, a couple lines up i make a array from Collection for cache

```php
return $permission->only('id', 'name', 'guard_name') + 
     ['roles' => $permission->roles->map->only('id', 'name', 'guard_name')->all()];
```
 `$permission->roles->map->only('id', 'name', 'guard_name')` returns array without `['pivot']`

Comment (drbyte) [reply src/PermissionRegistrar.php:138]: I guess if `$this->permissions` is never an array, it won't matter.

Comment (erikn69) [reply src/PermissionRegistrar.php:138]: Yes, if `$this->permissions` is a collection, it avoid that code, only with array makes `hydrate` for transform to a colection, It is to support transition, we can remove that condition on next major version, It works same as before(tested on mi applications, no `['pivot']`, minimal expression)

```php
array:358 [▼
  0 => array:4 [▼
    "id" => 197
    "name" => "view_audit"
    "guard_name" => "web"
    "roles" => array:3 [▼
      0 => array:3 [▼
        "id" => 0
        "name" => "System"
        "guard_name" => "web"
      ]
      1 => array:3 [▶]
      2 => array:3 [▶]
    ]
  ]
```

**Comment by erikn69** [comment]
@drbyte new idea, make cache even smaller using alias for field's names, i'm using ` ?? $item['id'], ?? $item['name']` for avoid breaks on everyone who already has 4.3.0

**Pull request merged**